http-client-core/src/main/java/io/micronaut/http/client/bind/DefaultHttpClientBinderRegistry.java

```diff
@@ -122,6 +124,17 @@ protected DefaultHttpClientBinderRegistry(ConversionService<?> conversionService
         byAnnotation.put(Body.class, (ClientArgumentRequestBinder<Object>) (context, uriContext, value, request) -> {
             request.body(value);
         });
+        byAnnotation.put(RequestBean.class, (ClientArgumentRequestBinder<Object>) (context, uriContext, value, request) -> {
+            BeanIntrospection<Object> introspection = BeanIntrospection.getIntrospection(context.getArgument().getType());
+            for (BeanProperty<Object, Object> beanProperty : introspection.getBeanProperties()) {
+                findArgumentBinder(beanProperty.asArgument()).ifPresent(binder -> {
+                    Object propertyValue = beanProperty.get(value);
+                    if (propertyValue != null) {
+                        binder.bind(ConversionContext.of(beanProperty.asArgument()), uriContext, propertyValue, request);
```

Comment:
This should be `context.with(beanProperty.asArgument())` so that errors are combined in the same context

Reply:
Done